client/me/help/help-contact/index.jsx

```diff
@@ -101,7 +101,8 @@ class HelpContact extends Component {
 	}
 
 	backToHelp = () => {
-		page( '/help' );
+		const searchParams = new URLSearchParams( window.location.search );
+		page( searchParams.get( 'redirect_to' ) ?? '/help' );
```

Comment:
[Should limit to relative urls only.](https://container-intelligent-chaum.calypso.live/help/contact?redirect_to=//example.com) 

```suggestion
		const redirectPath = searchParams.get( 'redirect_to' ) ?? '';
		if ( redirectPath.match( /^\/[^/]?/ ) ) {
			page( redirectPath );
			return;
		}
		page( '/help' );
		return;
```